tests/enterprise-contract/contract.go

```diff
@@ -253,5 +254,46 @@ var _ = framework.EnterpriseContractSuiteDescribe("Enterprise Contract E2E tests
 			Expect(err).NotTo(HaveOccurred())
 			Expect(reportLog).Should(ContainSubstring("Pipeline task 'build-container' uses an out of date task bundle"))
 		})
+
+		It("verifies the release policy: Task bundle references pinned to digest", func() {
+			secretName := fmt.Sprintf("unpinned-task-bundle-public-key%s", util.GenerateRandomString(10))
+			unpinnedTaskPublicKey := []byte("-----BEGIN PUBLIC KEY-----\n" +
+				"MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEPfwkY/ru2JRd6FSqIp7lT3gzjaEC\n" +
+				"EAg+paWtlme2KNcostCsmIbwz+bc2aFV+AxCOpRjRpp3vYrbS5KhkmgC1Q==\n" +
+				"-----END PUBLIC KEY-----")
+			GinkgoWriter.Println("Update public key to verify unpinned task image")
+			Expect(kubeController.CreateOrUpdateSigningSecret(unpinnedTaskPublicKey, secretName, namespace)).To(Succeed())
+			generator.PublicKey = fmt.Sprintf("k8s://%s/%s", namespace, secretName)
+			policy := ecp.EnterpriseContractPolicySpec{
+				Sources: policySource,
+				Configuration: &ecp.EnterpriseContractPolicyConfiguration{
+					Include: []string{"attestation_task_bundle.task_ref_bundles_pinned"},
+				},
+			}
+			Expect(kubeController.CreateOrUpdatePolicyConfiguration(namespace, policy)).To(Succeed())
+
+			generator.Image = "quay.io/redhat-appstudio-qe/enterprise-contract-tests:e2e-test-unpinned-task-bundle"
+			pr, err := kubeController.RunPipeline(generator, pipelineRunTimeout)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(kubeController.WatchPipelineRun(pr.Name, pipelineRunTimeout)).To(Succeed())
+
+			pr, err = kubeController.Tektonctrl.GetPipelineRun(pr.Name, pr.Namespace)
+			Expect(err).NotTo(HaveOccurred())
+
+			tr, err := kubeController.GetTaskRunStatus(fwk.AsKubeAdmin.CommonController.KubeRest(), pr, "verify-enterprise-contract")
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(tr.Status.TaskRunResults).Should(Or(
+				// TODO: delete the first option after https://issues.redhat.com/browse/RHTAP-810 is completed
+				ContainElements(tekton.MatchTaskRunResultWithJSONPathValue(constants.OldTektonTaskTestOutputName, "{$.result}", `["WARNING"]`)),
```

Comment:
I thought this field was removed now actually. Is it still there? Either way I think you can remove this line now.